### PR TITLE
Created CSharpWrapperApplication

### PR DESCRIPTION
**Pull request by HubertBalcerzak**

### CODE REVIEW

**Comment by loumalouomega** [comment]
Just one question, there is any way to check that it works?, I mean I assume that with Travis is going to be difficult to compile anything on C#, but maybe with appveyor

**Comment by loumalouomega** [comment]
I am meaning to add tests

**Comment by roigcarlo** [comment]
I don't think this applies to the standard application pipeline. If it does not break the rest of Kratos should be ok

File: applications/CSharpWrapperApplication/external_interface.cpp
Comment (RiccardoRossi) [inline applications/CSharpWrapperApplication/external_interface.cpp]: Being very picky: could u call this KratosCSharpWrapper or something of the sort?

**Comment by loumalouomega** [comment]
Of course this is not going to break the rest of the code, but the rest of the code can break the wrapper, so some day it can stop to work

File: applications/CSharpWrapperApplication/external_interface.cpp
Comment (RiccardoRossi) [inline applications/CSharpWrapperApplication/external_interface.cpp]: Also call 
       Interface --> CSharpInterface

not to close the keyword for others

File: applications/CSharpWrapperApplication/external_interface.h
Comment (RiccardoRossi) [inline applications/CSharpWrapperApplication/external_interface.h]: CSharpInterface

File: applications/CSharpWrapperApplication/external_interface_bindings.cpp
Comment (RiccardoRossi) [inline applications/CSharpWrapperApplication/external_interface_bindings.cpp:1]: @roigcarlo shouldn't this use the KRATOS_API stuff? your call...

File: applications/CSharpWrapperApplication/kratos_internals.cpp
Comment (RiccardoRossi) [inline applications/CSharpWrapperApplication/kratos_internals.cpp:4]: ugly. Shouldn't we call this from the interface?

File: applications/CSharpWrapperApplication/mesh_converter_old.cpp
Comment (RiccardoRossi) [inline applications/CSharpWrapperApplication/mesh_converter_old.cpp]: this is completely commented. do you need to upload this?

**Comment by RiccardoRossi** [comment]
Even hough i would be very much in agreement (in general) about the tests, i don't think it applies here, since we normally do not use C# in our pipeline. We should definitely not go throught the effort of mounting the whole compilation pipeline (which would involve unity) only for this...

that is...

i think we should allow this in even without tests (as i am telling, as a special case)

**Comment by jginternational** [comment]
Eternal glory => C# lovers!

**Review by RiccardoRossi — DISMISSED** [review]

**Comment by loumalouomega** [comment]
Just one question, can we rename the app to WrapperApplication and put there #2316?, maybe not now, but after merge

**Comment by HubertBalcerzak** [comment]
I agree about the rename, since it is possible to call this interface from a few different languages, not only C#

File: applications/CSharpWrapperApplication/mesh_converter_old.cpp
Comment (HubertBalcerzak) [reply applications/CSharpWrapperApplication/mesh_converter_old.cpp]: ups, my bad

File: applications/CSharpWrapperApplication/kratos_internals.cpp
Comment (HubertBalcerzak) [reply applications/CSharpWrapperApplication/kratos_internals.cpp:4]: No, I don't think this should be called from the interface. This modelpart is created internally

File: applications/CSharpWrapperApplication/external_interface_bindings.cpp
Comment (roigcarlo) [reply applications/CSharpWrapperApplication/external_interface_bindings.cpp:1]: Sorry totally forgot about this.
It depends, in general I don't think we need the `KRATOS_API()` stuff here, is not like this is gonna be used as a base class in another app, but if @Hubertus248 wants he can use it, the effect will be the same.

In any case, you already have that export defined as `KRATOS_API_EXPORT` if you don't want to define it again

**Review by RiccardoRossi — DISMISSED** [review]

**Review by RiccardoRossi — APPROVED** [review]

**Pull request merged**
